### PR TITLE
added instructions for building on windows

### PR DESCRIPTION
**Pull request by gianmaria**
you can now build nothing on windows!
I've added a build.bat file and a section in the readme.md for explanations.

Tested on Windows 10 64bit with VS2017 Comunity

### CODE REVIEW

File: build.bat [hunk 1]
```diff
@@ -0,0 +1,169 @@
+@echo off
```
Comment (rexim) [inline build.bat]: Sorry, the build process on all of the supported platforms should be done via [cmake](https://cmake.org/). I'm not accepting any custom build scripts, especially the ones that you have to modify for some reason (according to the instructions in README).

File: src/game/level/background.c [hunk 2]
```diff
@@ -7,6 +7,9 @@
 #include "system/lt.h"
 #include "system/nth_alloc.h"
 #include "system/log.h"
+#ifdef WINDOWS_OS
+#include <stdio.h> // for EOF macro
```
Comment (rexim) [inline src/game/level/background.c]: `stdio.h` exists on all of the platforms. Just include it without `#ifdef WINDOWS_OS`

File: src/game/level/boxes.c [hunk 3]
```diff
@@ -8,6 +8,9 @@
 #include "system/lt.h"
 #include "system/nth_alloc.h"
 #include "system/log.h"
+#ifdef WINDOWS_OS
+#include <stdio.h> // for EOF macro
```
Comment (rexim) [inline src/game/level/boxes.c]: `stdio.h` exists on all of the platforms. Just include it without `#ifdef WINDOWS_OS`

File: src/game/level/boxes.c [hunk 4]
```diff
@@ -33,7 +36,11 @@ Boxes *create_boxes_from_line_stream(LineStream *line_stream)
 
     if (sscanf(
             line_stream_next(line_stream),
+#ifdef X64_BUILD
```
Comment (rexim) [inline src/game/level/boxes.c]: Does `X64_BUILD` make exist on all of the platforms and compilers?

File: src/game/level/goals.c [hunk 5]
```diff
@@ -10,6 +10,9 @@
 #include "system/lt.h"
 #include "system/nth_alloc.h"
 #include "system/log.h"
+#ifdef WINDOWS_OS
+#include <stdio.h> // for EOF macro
```
Comment (rexim) [inline src/game/level/goals.c]: `stdio.h` exists on all of the platforms. Just include it without `#ifdef WINDOWS_OS`

File: src/game/level/labels.c [hunk 6]
```diff
@@ -8,6 +8,9 @@
 #include "system/lt.h"
 #include "system/nth_alloc.h"
 #include "system/log.h"
+#ifdef WINDOWS_OS
+#include <stdio.h> // for EOF macro
```
Comment (rexim) [inline src/game/level/labels.c]: `stdio.h` exists on all of the platforms. Just include it without `#ifdef WINDOWS_OS`

File: src/game/level_picker.c [hunk 7]
```diff
@@ -100,8 +100,9 @@ int level_picker_input(LevelPicker *level_picker,
 {
     trace_assert(level_picker);
     trace_assert(keyboard_state);
-    trace_assert(the_stick_of_joy);
-    return 0;
+    //trace_assert(the_stick_of_joy); // if you don't have a joystick the game will crash
```
Comment (rexim) [inline src/game/level_picker.c]: Please remove the dead commented code.

File: src/system/stacktrace.c [hunk 8]
```diff
@@ -3,7 +3,11 @@
 #ifdef __GNUC__
 #include <execinfo.h>
 #endif
-#include <unistd.h>
+#ifdef WINDOWS_OS
+    #include <io.h>
+#else
+    #include <unistd.h>
```
Comment (rexim) [inline src/system/stacktrace.c]: The macro should not work on Windows anyway. Please put `unistd.h` under `#ifdef __GNU__`.

File: src/game/level/boxes.c (hunk 4 above)
Comment (gianmaria) [reply src/game/level/boxes.c]: no actually no, i've defined it myself in the compilation process with the -D equivalent for the vs compiler cl.exe.
maybe i can use some define marcos from SDL2, if there's any?

File: build.bat (hunk 1 above)
Comment (gianmaria) [reply build.bat]: Unluckly i wasn't able to compile nothing with conan + cmake.
Although i've succeeded in generating the solution file for visual studio the compiled SDL2.lib and SDL2_Mixer.lib from cargo has some dependency that the linker can't figure it out.
Anyway i've updated the conanfile.txt with the dependencies and the readme.md with the instruction for building with windows.
:(

Comment (rexim) [reply build.bat]: @gianmaria Why didn't you file an issue complaining about bad documentation according to [Contribution Rules](https://github.com/tsoding/nothing/blob/master/CONTRIBUTING.md#issues)?

**Comment by rexim** [comment]
@gianmaria Since https://github.com/tsoding/nothing/pull/611#discussion_r244715947 was not addressed properly within 2 weeks, according to [Contribution Rules](https://github.com/tsoding/nothing/blob/master/CONTRIBUTING.md#pull-requests) I consider this PR neglected.

Thanks for submitting a PR. Feel free to try again at any time.

**Comment by rexim** [comment]
@gianmaria you might be interested in taking a look at the current Windows CI build. https://github.com/tsoding/nothing/blob/4a7bda399e3ccbe3ff62a388bca768715660bac5/appveyor.yml

**Pull request closed without merging**